File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/BigBullet.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

public class BigBullet extends Equipable {
	{
		sprite = "item-large_bullet";
	}

	@Override
	public void onEquip() {
		super.onEquip();
		this.owner.modifyStat("gun_use_time", -0.5f);
		this.owner.modifyStat("damage", 0.3f);
	}

	@Override
	public void onUnequip() {
		super.onUnequip();
		this.owner.modifyStat("gun_use_time", 0.5f);
		this.owner.modifyStat("damage", -0.3f);
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/BigBullet.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

class BigBullet : Equipable() {
	init {
		sprite = "item-large_bullet"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.modifyStat("gun_use_time", -0.5f)
		this.owner.modifyStat("damage", getChance() / 100)
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.modifyStat("gun_use_time", 0.5f)
		this.owner.modifyStat("damage", -getChance() / 100)
	}

	private fun getChance(): Float {
		return 10f + this.level * 20
	}

	override fun getMaxLevel(): Int {
		return 7
	}

	override fun getDescription(): String {
		return super.getDescription().replace("{PERCENT}", getChance().toInt().toString())
	}
}